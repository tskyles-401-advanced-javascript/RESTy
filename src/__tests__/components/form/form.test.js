import React from "react";
import Form from "../../../components/form/form";
import Enzyme, { mount, render, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
Enzyme.configure({ adapter: new Adapter() });

describe("<Form />", () => {
  it("is alive on start", () => {
    let app = shallow(<Form />);
    expect(app.find("form").exists()).toBeTruthy();
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
