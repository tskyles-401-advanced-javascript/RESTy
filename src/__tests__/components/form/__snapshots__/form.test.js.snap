// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders correctly 1`] = `
<form
  id="urlForm"
  onSubmit={[Function]}
>
  <input
    placeholder="URL"
  />
  <div
    className="selectors"
  >
    <a
      className="methods"
      href="/#"
      value="GET"
    >
      GET
    </a>
    <a
      className="methods"
      href="/#"
      value="POST"
    >
      POST
    </a>
    <a
      className="methods"
      href="/#"
      value="PUT"
    >
      PUT
    </a>
    <a
      className="methods"
      href="/#"
      value="PATCH"
    >
      PATCH
    </a>
    <a
      className="methods"
      href="/#"
      value="DELETE"
    >
      DELETE
    </a>
    <button>
      Go!
    </button>
  </div>
</form>
`;
